torch/_logging/_internal.py

```diff
@@ -0,0 +1,359 @@
+import logging
+import functools
+import os
+import re
+from importlib import __import__
+from typing import Dict, Set
+from dataclasses import dataclass, field
+
+DEFAULT_LOG_LEVEL = logging.WARN
+DEFAULT_FORMATTER = logging.Formatter(
+    "[%(asctime)s] %(name)s: [%(levelname)s] %(message)s"
+)
+LOG_ENV_VAR = "TORCH_LOGS"
+
+
+@dataclass
+class LogRegistry:
+    # shorthand name to log qualified name
+    log_alias_to_log_qname : Dict[str, str] = field(default_factory=dict)
+
+    artifact_log_qnames : Set[str] = field(default_factory=set)
+
+    artifact_names : Set[str] = field(default_factory=set)
```

Comment:
Do you need both of these? I would assume that artifact_log_qnames can be derived from artifact_names and log_alias_to_log_qname?=

Reply:
The issue is these are populated at different times, and you actually don't need an explicit mapping between these. `artifact_log_qname` is lazily initialized because it isn't populated until `getArtifactLogger` is called. `artifact_names` is initialized ahead of time by _registrations.py.

Reply:
I can add examples as you said, which should make this clearer